onmt/translate/beam.py

```diff
@@ -189,11 +189,11 @@ class GNMTGlobalScorer(object):
        beta (float):  coverage parameter
     """
 
-    def __init__(self, alpha, beta, cov_penalty, length_penalty):
-        self.alpha = alpha
-        self.beta = beta
-        penalty_builder = penalties.PenaltyBuilder(cov_penalty,
-                                                   length_penalty)
+    def __init__(self, opt):
```

Comment:
Not sure that's an improvement. It makes sense for high level classes to get the `opt` argument but other than that it is better to keep fully defined arguments IMO.